mod/groups/views/default/groups/edit/access.php

```diff
@@ -8,128 +8,92 @@
+	],
+	'label' => elgg_echo('groups:membership'),
+]);
 
-?>
-<div>
-	<label for="groups-membership"><?php echo elgg_echo("groups:membership"); ?></label><br />
-	<?php echo elgg_view("input/select", array(
-		"name" => "membership",
-		"id" => "groups-membership",
-		"value" => $membership,
-		"options_values" => array(
-			ACCESS_PRIVATE => elgg_echo("groups:access:private"),
-			ACCESS_PUBLIC => elgg_echo("groups:access:public"),
-		)
-	));
-	?>
-</div>
+if (elgg_get_plugin_setting('hidden_groups', 'groups') == 'yes') {
 
-<?php if (elgg_get_plugin_setting("hidden_groups", "groups") == "yes"): ?>
-	<div>
-		<label for="groups-vis"><?php echo elgg_echo("groups:visibility"); ?></label><br />
-		<?php
-		$visibility_options =  array(
-			ACCESS_PRIVATE => elgg_echo("groups:access:group"),
-			ACCESS_LOGGED_IN => elgg_echo("LOGGED_IN"),
-			ACCESS_PUBLIC => elgg_echo("PUBLIC"),
-		);
-		if (elgg_get_config("walled_garden")) {
-			unset($visibility_options[ACCESS_PUBLIC]);
-		}
-		
-		echo elgg_view("input/access", array(
-			"name" => "vis",
-			"id" => "groups-vis",
-			"value" => $visibility,
-			"options_values" => $visibility_options,
-			'entity' => $entity,
-			'entity_type' => 'group',
-			'entity_subtype' => '',
-		));
-		?>
-	</div>
-<?php endif; ?>
-
-<?php
+	$visibility_options = [
+		ACCESS_PRIVATE => elgg_echo('groups:access:group'),
+		ACCESS_LOGGED_IN => elgg_echo('LOGGED_IN'),
+		ACCESS_PUBLIC => elgg_echo('PUBLIC'),
+	];
+	if (elgg_get_config('walled_garden')) {
+		unset($visibility_options[ACCESS_PUBLIC]);
+	}
 
-$access_mode_params = array(
-	"name" => "content_access_mode",
-	"id" => "groups-content-access-mode",
-	"value" => $content_access_mode,
-	"options_values" => array(
-		ElggGroup::CONTENT_ACCESS_MODE_UNRESTRICTED => elgg_echo("groups:content_access_mode:unrestricted"),
-		ElggGroup::CONTENT_ACCESS_MODE_MEMBERS_ONLY => elgg_echo("groups:content_access_mode:membersonly"),
-	)
-);
+	echo elgg_view_input('access', [
+		'name' => 'vis',
+		'id' => 'groups-vis',
+		'value' => $visibility,
+		'options_values' => $visibility_options,
+		'entity' => $entity,
+		'entity_type' => 'group',
+		'entity_subtype' => '',
+		'label' => elgg_echo('groups:visibility'),
+	]);
+}
 
-if ($entity) {
+echo elgg_view_input('select', [
+	'name' => 'content_access_mode',
+	'id' => 'groups-content-access-mode',
+	'value' => $content_access_mode,
+	'options_values' => [
+		ElggGroup::CONTENT_ACCESS_MODE_UNRESTRICTED => elgg_echo('groups:content_access_mode:unrestricted'),
+		ElggGroup::CONTENT_ACCESS_MODE_MEMBERS_ONLY => elgg_echo('groups:content_access_mode:membersonly'),
+	],
+	'label' => elgg_echo('groups:content_access_mode'),
+	// Warn the user that changing the content access mode to more
+	// restrictive will not affect the existing group content
+	'help' => $entity && $entity->getContentAccessMode() == ElggGroup::CONTENT_ACCESS_MODE_UNRESTRICTED ? elgg_echo('groups:content_access_mode:warning') : '',
```

Comment:
BTW, what happens if help is set to `""` or `null`? Does it get a wrapper element?